Competition2017/src/competition/subsystems/shooter/SideShooterSubsystem.java

```diff
@@ -0,0 +1,166 @@
+    
+    private static Logger log = Logger.getLogger(ShooterSubsystem.class);
+    
+    private final XCANTalon masterMotor;
+    public final String side;
+    
+    // output telemetry properties
+    private final DoubleProperty shooterCurrentSpeed;
+    private final DoubleProperty shooterTargetSpeed;
+    private final DoubleProperty shooterOutputPower;
+    private final DoubleProperty shooterTalonError;
+    private final BooleanProperty atSpeedProp;
+    private final BooleanProperty enableShooterLogging;
+    
+    private final DoubleProperty p;
+    private final DoubleProperty i;
+    private final DoubleProperty d;
+    private final DoubleProperty f;
+    
+    private final DoubleProperty shooterSpeedThresh;
+    
+   
+    public SideShooterSubsystem(int motor, String side,WPIFactory factory, XPropertyManager propManager){
+        log.info("Creating" + side + " Shooter");
+        this.side = side;
+        
+        p = propManager.createPersistentProperty(side + "Shooter wheel P", 0);
+        i = propManager.createPersistentProperty(side + "Shooter wheel I", 0);
+        d = propManager.createPersistentProperty(side + "Shooter wheel D", 0);
+        f = propManager.createPersistentProperty(side + "Shooter wheel F", 0);
+        
+        // Reading through the manual, it looks like only a few things support human units. It might be easier to just keep everything in native units per 100ms,
+        // which is what the Talon uses for all its calculations.
+        // I call this Ticks per Deciseconds, or TPC
+        shooterSpeedThresh = propManager.createPersistentProperty(side + "Shooter nominal speed thresh (TPC)", 1);
+        shooterCurrentSpeed = propManager.createEphemeralProperty(side + "Shooter current speed (TPC)", 0);
+        shooterTargetSpeed = propManager.createEphemeralProperty(side + "Shooter goal speed (TPC)", 0);
+        shooterOutputPower = propManager.createEphemeralProperty(side + "Shooter voltage", 0);
+        atSpeedProp = propManager.createEphemeralProperty("Is" + side + "Shooter at speed?", false);
+        shooterTalonError = propManager.createEphemeralProperty(side + "Shooter speed error", 0);
+        enableShooterLogging = propManager.createEphemeralProperty("Is Shooter logging info?", false);
+        
+        this.masterMotor = factory.getCANTalonSpeedController(1);
+        setUpMotorTeam(masterMotor);
+        masterMotor.createTelemetryProperties(side + "Shooter master", propManager);
+    }
+    
+    private void setUpMotorTeam(XCANTalon master) {
+        // TODO: Check faults and voltage/temp/current
+        log.info("Configuring" + side + "motor");
+        // Master config
+        master.setFeedbackDevice(FeedbackDevice.QuadEncoder);
+
+        master.reverseSensor(true);
+        master.setInverted(true);
+        log.info("Configuring voltages");
+        master.configNominalOutputVoltage(0,  -0);
+        master.configPeakOutputVoltage(12, -12);
```

Comment:
Sounds like a good protection! If we do that, let's surround that code with a big comment in case we start wondering "why the heck is the power so low?"